cmd/terramate/cli/cli.go

```diff
@@ -628,3 +634,116 @@ func newGit(basedir string, inheritEnv bool, checkrepo bool) (*git.Git, error) {
 
 	return g, nil
 }
+
+func lookupProject(wd string) (found bool, prj project, err error) {
+	prj = project{
+		wd: wd,
+	}
+	gw, err := newGit(wd, false, false)
+	if err == nil {
+		gitdir, err := gw.Root()
+		if err == nil {
+			gitabs, err := filepath.Abs(gitdir)
+			if err != nil {
+				return false, project{}, fmt.Errorf("getting absolute path of %q: %w", gitdir, err)
+			}
+
+			gitabs, err = filepath.EvalSymlinks(gitabs)
+			if err != nil {
+				return false, project{}, fmt.Errorf("failed evaluating symlinks of %q: %w",
+					gitabs, err)
+			}
+
+			root := filepath.Dir(gitabs)
+			_, cfg, err := config.TryLoadRootConfig(root)
+			if err != nil {
+				return false, project{}, err
+			}
+
+			prj.isRepo = true
+			prj.cfg = cfg
+			prj.root = root
+
+			return true, prj, nil
+		}
+	}
+
+	dir := wd
+
+	for {
```

Comment:
```suggestion
	for dir != "/" {
```

Debatable but I usually prefer the stopping criteria on the loop declaration (if possible). In this case it seems possible, but maybe Im wrong, disregard at will

Reply:
yeah me too but in this case it needs to check at `/` as well and if the config is still not there then we break the loop. It'd be a perfect case for a `do { ... } while(dir != "/")` haha

Reply:
Ah got it, I missed this :+1: